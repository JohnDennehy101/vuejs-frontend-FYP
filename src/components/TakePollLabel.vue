<template>
  <label class="checkbox-label" :for="id"
    >{{ formattedStartDate }} - {{ formattedEndDate }}</label
  >
</template>

<script>
export default {
  props: ["optionId", "optionStartDate", "optionEndDate"],
  data() {
    return {
      start: this.optionStartDate,
      end: this.optionEndDate,
      id: this.optionId,
    };
  },
  computed: {
    formattedStartDate: function () {
      let date = new Date(this.start);

      let month = date.getMonth();
      let months = [
        "January",
        "February",
        "March",
        "April",
        "May",
        "June",
        "July",
        "August",
        "September",
        "October",
        "November",
        "December",
      ];

      let formattedDate =
        ("0" + date.getDate()).slice(-2) +
        "-" +
        (months[month] + "-" + date.getFullYear() + " ");

      return formattedDate;
    },
    formattedEndDate: function () {
      let date = new Date(this.end);
      let month = date.getMonth();
      let months = [
        "January",
        "February",
        "March",
        "April",
        "May",
        "June",
        "July",
        "August",
        "September",
        "October",
        "November",
        "December",
      ];

      let formattedDate =
        ("0" + date.getDate()).slice(-2) +
        "-" +
        (months[month] + "-" + date.getFullYear() + " ");

      return formattedDate;
    },
  },
};
</script>

<style scoped lang="scss">
.checkbox-label {
  display: inline-block;
  margin-bottom: 0;
}
</style>
