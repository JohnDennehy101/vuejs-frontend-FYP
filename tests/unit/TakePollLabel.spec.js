import { mount, shallowMount } from "@vue/test-utils";
import TakePollLabel from "@/components/TakePollLabel.vue";

describe("TakePollLabel.vue", () => {
  afterEach(() => {
        jest.resetAllMocks();
    });
  it("should render correctly", () => {
    const wrapper = shallowMount(TakePollLabel);
    expect(wrapper).toMatchSnapshot();
  });

  it("checks that poll option dates are rendered correctly", async () => {
    const wrapper = mount(TakePollLabel, {
      props: {
        start: new Date(),
        end: new Date(),
        id: "1"
      },
    });

    expect(wrapper.find("label")).toBeDefined();
  });

  it("checks computed properties are defined", async () => {
    mount(TakePollLabel, {
       props: {
        startDate: new Date(),
        endDate: new Date(),
        id: "1"
      },
    });
    expect(TakePollLabel.computed.formattedStartDate()).toBeDefined();
    expect(TakePollLabel.computed.formattedEndDate()).toBeDefined();
  });
});
