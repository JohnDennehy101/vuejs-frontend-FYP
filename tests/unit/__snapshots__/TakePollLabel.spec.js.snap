// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TakePollLabel.vue should render correctly 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "name": "VTU_ROOT",
      "render": [Function],
      "setup": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <label
        class="checkbox-label"
      >
        aN-undefined-NaN  - aN-undefined-NaN 
      </label>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
        },
        "transition-group": Object {
          "name": "transition-group",
        },
      },
      "config": Object {
        "compilerOptions": Object {},
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "emitsCache": WeakMap {},
      "mixins": Array [
        Object {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": Object {},
      "reload": [Function],
    },
    "_instance": Object {
      "a": null,
      "accessCache": Object {},
      "appContext": Object {
        "app": [Circular],
        "components": Object {
          "transition": Object {
            "name": "transition",
          },
          "transition-group": Object {
            "name": "transition-group",
          },
        },
        "config": Object {
          "compilerOptions": Object {},
          "errorHandler": undefined,
          "globalProperties": Object {},
          "isNativeTag": [Function],
          "optionMergeStrategies": Object {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": Object {},
        "emitsCache": WeakMap {},
        "mixins": Array [
          Object {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": Object {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": Object {},
      "attrsProxy": null,
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": Object {
        "VTU_COMPONENT": Object {
          "end": undefined,
          "formattedEndDate": "aN-undefined-NaN ",
          "formattedStartDate": "aN-undefined-NaN ",
          "id": undefined,
          "optionEndDate": undefined,
          "optionId": undefined,
          "optionStartDate": undefined,
          "start": undefined,
        },
      },
      "da": null,
      "data": Object {},
      "devtoolsRawSetupState": Object {
        "VTU_COMPONENT": RefImpl {
          "__v_isRef": true,
          "__v_isShallow": false,
          "_rawValue": Object {
            "end": undefined,
            "formattedEndDate": "aN-undefined-NaN ",
            "formattedStartDate": "aN-undefined-NaN ",
            "id": undefined,
            "optionEndDate": undefined,
            "optionId": undefined,
            "optionStartDate": undefined,
            "start": undefined,
          },
          "_value": Object {
            "end": undefined,
            "formattedEndDate": "aN-undefined-NaN ",
            "formattedStartDate": "aN-undefined-NaN ",
            "id": undefined,
            "optionEndDate": undefined,
            "optionId": undefined,
            "optionStartDate": undefined,
            "start": undefined,
          },
          "dep": undefined,
        },
      },
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": Array [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": Object {},
      "propsDefaults": Object {},
      "propsOptions": Array [
        Object {},
        Array [],
      ],
      "provides": Object {},
      "proxy": Object {
        "VTU_COMPONENT": Object {
          "end": undefined,
          "formattedEndDate": "aN-undefined-NaN ",
          "formattedStartDate": "aN-undefined-NaN ",
          "id": undefined,
          "optionEndDate": undefined,
          "optionId": undefined,
          "optionStartDate": undefined,
          "start": undefined,
        },
      },
      "refs": Object {
        "VTU_COMPONENT": Object {
          "end": undefined,
          "formattedEndDate": "aN-undefined-NaN ",
          "formattedStartDate": "aN-undefined-NaN ",
          "id": undefined,
          "optionEndDate": undefined,
          "optionId": undefined,
          "optionStartDate": undefined,
          "start": undefined,
        },
      },
      "render": [Function],
      "renderCache": Array [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "_active": true,
        "cleanups": Array [],
        "detached": true,
        "effects": Array [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
        "parent": undefined,
      },
      "setupContext": null,
      "setupState": Object {
        "VTU_COMPONENT": Object {
          "end": undefined,
          "formattedEndDate": "aN-undefined-NaN ",
          "formattedStartDate": "aN-undefined-NaN ",
          "id": undefined,
          "optionEndDate": undefined,
          "optionId": undefined,
          "optionStartDate": undefined,
          "start": undefined,
        },
      },
      "slots": Object {},
      "slotsProxy": null,
      "sp": null,
      "subTree": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": Object {
          "a": null,
          "accessCache": Object {
            "@@__IMMUTABLE_ITERABLE__@@": 0,
            "@@__IMMUTABLE_RECORD__@@": 0,
            "__v_isReadonly": 0,
            "__v_isRef": 0,
            "__v_isShallow": 0,
            "__v_raw": 0,
            "__v_skip": 4,
            "_isMockFunction": 0,
            "constructor": 0,
            "end": 2,
            "formattedEndDate": 4,
            "formattedStartDate": 4,
            "id": 2,
            "isVueInstance": 0,
            "optionEndDate": 3,
            "optionId": 3,
            "optionStartDate": 3,
            "start": 2,
            "toJSON": 0,
            Symbol(Symbol.toStringTag): 0,
          },
          "appContext": Object {
            "app": [Circular],
            "components": Object {
              "transition": Object {
                "name": "transition",
              },
              "transition-group": Object {
                "name": "transition-group",
              },
            },
            "config": Object {
              "compilerOptions": Object {},
              "errorHandler": undefined,
              "globalProperties": Object {},
              "isNativeTag": [Function],
              "optionMergeStrategies": Object {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": Object {},
            "emitsCache": WeakMap {},
            "mixins": Array [
              Object {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": Object {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": Object {},
          "attrsProxy": null,
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": Object {},
          "ctx": Object {
            "end": undefined,
            "formattedEndDate": "aN-undefined-NaN ",
            "formattedStartDate": "aN-undefined-NaN ",
            "id": undefined,
            "optionEndDate": undefined,
            "optionId": undefined,
            "optionStartDate": undefined,
            "start": undefined,
          },
          "da": null,
          "data": Object {
            "end": undefined,
            "id": undefined,
            "start": undefined,
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": null,
          "exposed": null,
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": Object {
            "optionEndDate": undefined,
            "optionId": undefined,
            "optionStartDate": undefined,
          },
          "propsDefaults": Object {},
          "propsOptions": Array [
            Object {
              "optionEndDate": Object {},
              "optionId": Object {},
              "optionStartDate": Object {},
            },
            Array [],
          ],
          "provides": Object {},
          "proxy": Object {
            "end": undefined,
            "formattedEndDate": "aN-undefined-NaN ",
            "formattedStartDate": "aN-undefined-NaN ",
            "id": undefined,
            "optionEndDate": undefined,
            "optionId": undefined,
            "optionStartDate": undefined,
            "start": undefined,
          },
          "refs": Object {},
          "render": [Function],
          "renderCache": Array [],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "_active": true,
            "cleanups": Array [],
            "detached": true,
            "effects": Array [
              ReactiveEffect {
                "active": true,
                "computed": ComputedRefImpl {
                  "__v_isReadonly": false,
                  "__v_isRef": true,
                  "_cacheable": true,
                  "_dirty": false,
                  "_setter": [Function],
                  "_value": "aN-undefined-NaN ",
                  "dep": Set {
                    ReactiveEffect {
                      "active": true,
                      "allowRecurse": true,
                      "deps": Array [
                        Set {
                          [Circular],
                        },
                        [Circular],
                        Set {
                          [Circular],
                        },
                      ],
                      "fn": [Function],
                      "onTrack": undefined,
                      "onTrigger": undefined,
                      "parent": undefined,
                      "scheduler": [Function],
                    },
                  },
                  "effect": [Circular],
                },
                "deps": Array [
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "parent": undefined,
                "scheduler": [Function],
              },
              ReactiveEffect {
                "active": true,
                "computed": ComputedRefImpl {
                  "__v_isReadonly": false,
                  "__v_isRef": true,
                  "_cacheable": true,
                  "_dirty": false,
                  "_setter": [Function],
                  "_value": "aN-undefined-NaN ",
                  "dep": Set {
                    ReactiveEffect {
                      "active": true,
                      "allowRecurse": true,
                      "deps": Array [
                        Set {
                          [Circular],
                        },
                        Set {
                          [Circular],
                        },
                        [Circular],
                      ],
                      "fn": [Function],
                      "onTrack": undefined,
                      "onTrigger": undefined,
                      "parent": undefined,
                      "scheduler": [Function],
                    },
                  },
                  "effect": [Circular],
                },
                "deps": Array [
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "parent": undefined,
                "scheduler": [Function],
              },
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": Array [
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
            "parent": undefined,
          },
          "setupContext": null,
          "setupState": Object {},
          "slots": Object {},
          "slotsProxy": null,
          "sp": null,
          "subTree": Object {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": "aN-undefined-NaN  - aN-undefined-NaN ",
            "component": null,
            "ctx": [Circular],
            "dirs": null,
            "dynamicChildren": Array [],
            "dynamicProps": Array [
              "for",
            ],
            "el": <label
              class="checkbox-label"
            >
              aN-undefined-NaN  - aN-undefined-NaN 
            </label>,
            "key": null,
            "patchFlag": 9,
            "props": Object {
              "class": "checkbox-label",
              "for": undefined,
            },
            "ref": null,
            "scopeId": null,
            "shapeFlag": 9,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "label",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": Object {
            "components": Object {},
            "computed": Object {
              "formattedEndDate": [Function],
              "formattedStartDate": [Function],
            },
            "data": [Function],
            "props": Array [
              "optionId",
              "optionStartDate",
              "optionEndDate",
            ],
            "render": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "ctx": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <label
          class="checkbox-label"
        >
          aN-undefined-NaN  - aN-undefined-NaN 
        </label>,
        "key": null,
        "patchFlag": 0,
        "props": Object {
          "ref": "VTU_COMPONENT",
        },
        "ref": Object {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "components": Object {},
          "computed": Object {
            "formattedEndDate": [Function],
            "formattedStartDate": [Function],
          },
          "data": [Function],
          "props": Array [
            "optionId",
            "optionStartDate",
            "optionEndDate",
          ],
          "render": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": Object {
        "name": "VTU_ROOT",
        "render": [Function],
        "setup": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": Object {
          "app": [Circular],
          "components": Object {
            "transition": Object {
              "name": "transition",
            },
            "transition-group": Object {
              "name": "transition-group",
            },
          },
          "config": Object {
            "compilerOptions": Object {},
            "errorHandler": undefined,
            "globalProperties": Object {},
            "isNativeTag": [Function],
            "optionMergeStrategies": Object {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": Object {},
          "emitsCache": WeakMap {},
          "mixins": Array [
            Object {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": Object {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "ctx": null,
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <label
          class="checkbox-label"
        >
          aN-undefined-NaN  - aN-undefined-NaN 
        </label>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "name": "VTU_ROOT",
          "render": [Function],
          "setup": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "compilerOptions": Object {},
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "runWithContext": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.3.4",
  },
  "__setProps": [Function],
  "cleanUpCallbacks": Array [
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
  ],
  "componentVM": Object {
    "end": undefined,
    "formattedEndDate": "aN-undefined-NaN ",
    "formattedStartDate": "aN-undefined-NaN ",
    "id": undefined,
    "optionEndDate": undefined,
    "optionId": undefined,
    "optionStartDate": undefined,
    "start": undefined,
  },
  "isDisabled": [Function],
  "rootVM": Object {
    "VTU_COMPONENT": Object {
      "end": undefined,
      "formattedEndDate": "aN-undefined-NaN ",
      "formattedStartDate": "aN-undefined-NaN ",
      "id": undefined,
      "optionEndDate": undefined,
      "optionId": undefined,
      "optionStartDate": undefined,
      "start": undefined,
    },
  },
  "wrapperElement": <label
    class="checkbox-label"
  >
    aN-undefined-NaN  - aN-undefined-NaN 
  </label>,
}
`;
